plugins/dead_code_elimination/dead_code_elimination_main.ml

```diff
@@ -73,27 +75,63 @@ let propagate_consts sub =
       Map.fold vars ~init:exp ~f:(fun ~key:pat ~data:rep ->
           Exp.substitute Bil.(var pat) rep)))
 
-let rec process proj =
-  let prog = Project.program proj in
+let free_vars (_,ssa) = free_vars ssa
+let compute_dead free (_,ssa) = compute_dead free ssa
+let is_alive dead t = not (Set.mem dead (Term.tid t))
+let live_def dead blk = Term.filter def_t ~f:(is_alive dead) blk
+let live_phi dead blk = Term.filter phi_t ~f:(is_alive dead) blk
+
+let clean dead (sub,ssa) =
+  Term.map blk_t sub ~f:(live_def dead),
+  Term.map blk_t ssa ~f:(fun b -> live_def dead b |> live_phi dead)
+
+let rec run subs =
```

Comment:
Translate the `run` to a function, that takes a _list_ of functions in the SSA form and the initial set of dead terms, and returns the final (maximal) set of dead terms. E.g.,

```ocaml

let rec run subs dead = 
   let free = ... in
   let dead' = .. in
   if Set.length dead' > Set.length dead 
   then run (clean dead' subs) dead'
   else dead
```